### PR TITLE
UI-header

### LINKED ISSUE #14: Build out header UI
Header UI should be as per mocks, responsive for both mobile and desktop
[mobile mock](https://drive.google.com/file/d/1UJFMzfhAcZf2cRA9wRgmsmC6Ce8-vG9u/view?usp=sharing)
[desktop mock](https://drive.google.com/file/d/15iz1SBpZceGvp79EJTJbmoUld85inOYb/view?usp=sharing)
[Google Doc](https://docs.google.com/document/d/1HSpsMwOwIRCP3S4_U3NVPEYuEVA1c57BDaJocpbUyj4/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by nickmeinhold**
EDIT: @T-Davis below is just from the last commit, the PR changes are the dot points below plus some scattered stuff in the commit history, I'm feeling lazy so my plan is to leave the PR comment as is but go through the changes and make comments, very happy to do better if you wanna pull me up on it though 🙂 

- added an extension on BoxConstraints to DRY up screen size checks
- removed null checks on SealDocs extension again... I don't get what's
happening, QueryDocumentSnapshot's data()
function definitely returns a non-null
type as far as I can see... I guess we'll
see what the CI says...
- I was nosing about in footer.dart and thought the Footer builder
would be quite neat with functional
syntax, I think I'm slightly obsessed...
feel free to tell me to butt out!

### CODE REVIEW

File: lib/UI/components/footer.dart [hunk 1]
```diff
@@ -6,15 +6,11 @@ double largeFont = 24;
...
-  Widget build(BuildContext context) {
-    return LayoutBuilder(builder: (context, constraints) {
-      if (constraints.maxWidth < 650) {
-        return FooterLayoutMobile();
-      } else {
-        return FooterLayoutDesktop();
-      }
-    });
-  }
+  Widget build(BuildContext context) => LayoutBuilder(
```
Comment (nickmeinhold) [inline lib/UI/components/footer.dart:9]: This was me butting in with my functional obsession - happy to undo!

File: lib/UI/components/footer.dart [hunk 2]
```diff
@@ -31,6 +27,19 @@ class FooterLayoutMobile extends StatelessWidget {
 }
 
 class FooterLayoutDesktop extends StatelessWidget {
+  @override
```
Comment (nickmeinhold) [inline lib/UI/components/footer.dart]: I didn't touch any of this - I think maybe it got caught in the merge... any thoughts?

Comment (nickmeinhold) [reply lib/UI/components/footer.dart]: I might have accidentally reverted something?

File: macos/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -258,6 +258,23 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
+		32C0458E291902DEE04D14A7 /* [CP] Copy Pods Resources */ = {
```
Comment (nickmeinhold) [inline macos/Runner.xcodeproj/project.pbxproj:261]: Just XCode doing it's autogenerated shiz I believe

File: lib/UI/components/footer.dart (hunk 1 above)
Comment (T-Davis) [reply lib/UI/components/footer.dart:9]: I like it!

File: lib/UI/components/footer.dart (hunk 2 above)
Comment (T-Davis) [reply lib/UI/components/footer.dart]: Oh, that is strange

**Review by T-Davis — APPROVED** [review]
Thanks again mate! All your work is gold 🏅

**Pull request merged**